commit 1aad063736f9ccaf7cdc7dab3db5ce5a371579b8 (refs/changes/24/4724/3)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2018-09-07T01:20:20+00:00 (1 year, 1 month ago)
    
    TRITON-707 CNS crash on access to query.src because redis claim() took too long, remote closed TCP connection
    Reviewed by: Robert Mustacchi <rm@joyent.com>

diff --git a/lib/dns-server.js b/lib/dns-server.js
index c4c323b..ce099cf 100644
--- a/lib/dns-server.js
+++ b/lib/dns-server.js
@@ -236,6 +236,12 @@ DNSServer.prototype.handleQuery = function (q, cb) {
 
 	q.log.trace('begin query');
 
+	if (q.src === undefined || q.src === null) {
+		q.log.warn('dropping query from already-closed connection');
+		cb();
+		return;
+	}
+
 	var name = q.name().toLowerCase();
 	var type = q.type();
 	var addr = q.src.address;
